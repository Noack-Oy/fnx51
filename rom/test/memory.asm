
; ************************************
; * Memory Management Test Program *
; ************************************

; Interactive test:
; d <addr> <size>; 	- dump memory
; g <size>;		- getmem
; f <addr> <size>;	- freemem


; **********************
; * Header Definitions *
; **********************
.inc ../global/variables.equ
.inc ../global/sfr.equ
.inc ../serial/sfr.equ


; *************
; * Main Code *
; *************

.org 0
.inc ../global/init.inc

; *** test/memory.asm ***

	acall	serial_init

	; use xram
	mov	a,	auxr
	anl	a,	0xe1	; clear extram, xrs0-2
	orl	a,	0x10	; set xrs2 (size 1792 bytes)
	mov	auxr,	a


loop:
	acall	read_char

	cjne	a,	#'d',	__1
	acall	read_hex_32
	mov	stream_in,	r0
	mov	stream_in+1,	r1
	acall	read_hex_32
	mov	a,	r0
	mov	r0,	stream_in
	mov	r1,	stream_in+1
	mov	r2,	#0
	mov	r3,	#0
	acall	dump
	sjmp	loop
__1:
	cjne	a,	#'g',	__2
	acall	read_hex_32
	acall	getmem
	acall	print_hex_32
	sjmp	loop
__2:
	cjne	a,	#'f',	__3
	acall	read_hex_32
	mov	a,	r0
	mov	r4,	a
	mov	a,	r1
	mov	r5,	a
	acall	read_hex_32
	mov	a,	r0
	mov	r2,	a
	mov	a,	r1
	mov	r3,	a
	mov	a,	r4
	mov	r0,	a
	mov	a,	r5
	mov	r1,	a
	acall	freemem
__3:
	mov	a,	#'?'
	acall	print_char
	sjmp	loop


; *********************
; * Library Functions *
; *********************
.inc ../serial/init.inc
.inc ../serial/tx.inc
.inc ../serial/rx.inc
.inc ../read/char.inc
.inc ../read/hex.inc
.inc ../print/char.inc
.inc ../print/text.inc
.inc ../print/hex.inc
.inc ../util/xch.inc
.inc ../util/regbank.inc
.inc ../util/dump.inc
.inc ../memory/getmem.inc
.inc ../memory/freemem.inc

.equ panic_out, serial_tx
.inc ../util/panic.inc
