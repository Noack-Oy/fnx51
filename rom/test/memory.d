memory.lst memory.hex: memory.asm ../global/variables.equ ../global/sfr.equ ../serial/sfr.equ ../global/init.inc ../serial/init.inc ../serial/tx.inc ../serial/rx.inc ../read/char.inc ../read/hex.inc ../print/char.inc ../print/text.inc ../print/hex.inc ../util/xch.inc ../util/regbank.inc ../util/dump.inc ../stream/xram_read.inc ../memory/getmem.inc ../memory/freemem.inc ../util/panic.inc
