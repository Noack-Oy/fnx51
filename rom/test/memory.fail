BOOTLOADER
> U
< U
> :020000050B00EE
< :020000050B00EE7B.\r\n
WRITE: memory.hex
> :100000007581409000B18582308583319000A9854B
< :100000007581409000B18582308583319000A9854B.\r\n
> :100010008232858333119FE58E55E14510F58E9030
< :100010008232858333119FE58E55E14510F58E9030.\r\n
> :1000200000008582388583397404F0A3E4F0A3745A
< :1000200000008582388583397404F0A3E4F0A3745A.\r\n
> :10003000FCF0A37406F0A3E4F0A3F0A374FCF0A317
< :10003000FCF0A37406F0A3E4F0A3F0A374FCF0A317.\r\n
> :100040007406F0A311B9B4642611BE883489351141
< :100040007406F0A311B9B4642611BE883489351141.\r\n
> :10005000BEE8A834A9357A007B00C030C0319002D8
< :10005000BEE8A834A9357A007B00C030C0319002D8.\r\n
> :100060005E85823085833131A2D031D03080D5B4E5
< :100060005E85823085833131A2D031D03080D5B4E5.\r\n
> :10007000670811BE51813158801BB4661411BEE867
< :10007000670811BE51813158801BB4661411BEE867.\r\n
> :10008000FCE9FD11BEE8FAE9FBECF8EDF9518880D6
< :10008000FCE9FD11BEE8FAE9FBECF8EDF9518880D6.\r\n
> :1000900004743F311D740D311D740A311D80A57526
< :1000900004743F311D740D311D740A311D80A57526.\r\n
> :1000A0009852759ABF759B1E223099FDC299F59999
< :1000A0009852759ABF759B1E223099FDC299F59999.\r\n
> :1000B000223098FDC298E59922C030C03122C0E0BC
< :1000B000223098FDC298E59922C030C03122C0E0BC.\r\n
> :1000C000ECC0E0EDC0E0E4F8F9FAFB11B9C39430FC
< :1000C000ECC0E0EDC0E0E4F8F9FAFB11B9C39430FC.\r\n
> :1000D0004042B40A00400D94114039240A54DFB460
< :1000D0004042B40A00400D94114039240A54DFB460.\r\n
> :1000E00010005030FCE8C4FD54F04CF8ED540FFC07
< :1000E00010005030FCE8C4FD54F04CF8ED540FFC07.\r\n
> :1000F000E9C4FD54F04CF9ED540FFCEAC4FD54F092
< :1000F000E9C4FD54F04CF9ED540FFCEAC4FD54F092.\r\n
> :100100004CFAED540FFCEBC4FD54F04CFBED540FD6
< :100100004CFAED540FFCEBC4FD54F04CFBED540FD6.\r\n
> :10011000700280B7D0E0FDD0E0FCD0E022C032C059
< :10011000700280B7D0E0FDD0E0FCD0E022C032C059.\r\n
> :100120003322C0E0E8C0E0E4F8936006311D08E83F
< :100120003322C0E0E8C0E0E4F8936006311D08E83F.\r\n
> :1001300070F7D0E0F8D0E022C4313CC4C0E0540FE6
< :1001300070F7D0E0F8D0E022C4313CC4C0E0540FE6.\r\n
> :100140002490D43440D44420311DD0E022C0E0E9D2
< :100140002490D43440D44420311DD0E022C0E0E9D2.\r\n
> :100150003138E83138D0E022C0E0EB3138EA3138CC
< :100150003138E83138D0E022C0E0EB3138EA3138CC.\r\n
> :10016000E93138E83138D0E022C8CCC8C9CDC9CA95
< :10016000E93138E83138D0E022C8CCC8C9CDC9CA95.\r\n
> :10017000CECACBCFCB22C0E0E5D05418B418025180
< :10017000CECACBCFCB22C0E0E5D05418B418025180.\r\n
> :100180008FE5D02408F5D0541824F8F8D0E022C028
< :100180008FE5D02408F5D0541824F8F8D0E022C028.\r\n
> :10019000E0E5D054187002518FE5D024F8F5D0D0A6
< :10019000E0E5D054187002518FE5D024F8F5D0D0A6.\r\n
> :1001A000E022C0F0C0E0F5F0ECC0E0EDC0E0EEC051
< :1001A000E022C0F0C0E0F5F0ECC0E0EDC0E0EEC051.\r\n
> :1001B000E0EFC0E03169E581F82410F58179101194
< :1001B000E0EFC0E03169E581F82410F58179101194.\r\n
> :1001C000B908F6D5F003198002D9F43169315831F4
< :1001C000B908F6D5F003198002D9F43169315831F4.\r\n
> :1001D00069743A311DE824F029F87A1008EA5403CA
< :1001D00069743A311DE824F029F87A1008EA5403CA.\r\n
> :1001E00070047420311DEAD39950087420311D31F8
< :1001E00070047420311DEAD39950087420311D31F8.\r\n
> :1001F0001D8003E631387420311DDAE0E824F0F880
< :1001F0001D8003E631387420311DDAE0E824F0F880.\r\n
> :100200007420311D747C311D7410C399F9FA08E60D
< :100200007420311D747C311D7410C399F9FA08E60D.\r\n
> :10021000B420004005B47F004002742E311DDAEE98
< :10021000B420004005B47F004002742E311DDAEE98.\r\n
> :10022000747C311D740D311D740A311DE824F0F801
< :10022000747C311D740D311D740A311DE824F0F801.\r\n
> :10023000EC29FCED3400FDEE3400FEEF3400FFE568
< :10023000EC29FCED3400FDEE3400FEEF3400FFE568.\r\n
> :10024000F0600221BDE58124F0F5813169D0E0FF45
< :10024000F0600221BDE58124F0F5813169D0E0FF45.\r\n
> :10025000D0E0FED0E0FDD0E0FCD0E0D0F022C08EB7
< :10025000D0E0FED0E0FDD0E0FCD0E0D0F022C08EB7.\r\n
> :10026000C082C083E58E55E14510F58E85348285C8
< :10026000C082C083E58E55E14510F58E85348285C8.\r\n
> :100270003583E0A3858234858335D083D082D08EC8
< :100270003583E0A3858234858335D083D082D08EC8.\r\n
> :1002800022C0A8C2AFD0A822C0A8C2AFD0A822C2A4
< :1002800022C0A8C2AFD0A822C0A8C2AFD0A822C2A4.\r\n
> :10029000AF9000A98582328583339002A93122D0A4
< :10029000AF9000A98582328583339002A93122D0A4.\r\n
> :1002A000E03138D0E0313880FE0D0A2021212120B4
< :1002A000E03138D0E0313880FE0D0A2021212120B4.\r\n
> :0802B00070616E6963204000DB
< :0802B00070616E6963204000DB.\r\n
> :00000001FF
< :00000001FF.\r\n
RESET
> d0;10;
00000000: 04 00 fc 06  00 00 fc 06  00 00 00 20  08 02 14 02  |........... ....|
> g123;
00000123
