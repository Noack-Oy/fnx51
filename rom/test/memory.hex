:100000007581409000B18582308583319000A9854B
:100010008232858333119FE58E55E14510F58E9030
:1000200000008582388583397404F0A3E4F0A3745A
:10003000FCF0A37406F0A3E4F0A3F0A374FCF0A317
:100040007406F0A311B9B4642611BE883489351141
:10005000BEE8A834A9357A007B00C030C0319002D8
:100060005E85823085833131A2D031D03080D5B4E5
:10007000610811BE51813158801BB4721411BEE861
:10008000FCE9FD11BEE8FAE9FBECF8EDF9714680F8
:1000900004743F311D740D311D740A311D80A57526
:1000A0009852759ABF759B1E223099FDC299F59999
:1000B000223098FDC298E59922C030C03122C0E0BC
:1000C000ECC0E0EDC0E0E4F8F9FAFB11B9C39430FC
:1000D0004042B40A00400D94114039240A54DFB460
:1000E00010005030FCE8C4FD54F04CF8ED540FFC07
:1000F000E9C4FD54F04CF9ED540FFCEAC4FD54F092
:100100004CFAED540FFCEBC4FD54F04CFBED540FD6
:10011000700280B7D0E0FDD0E0FCD0E022C032C059
:100120003322C0E0E8C0E0E4F8936006311D08E83F
:1001300070F7D0E0F8D0E022C4313CC4C0E0540FE6
:100140002490D43440D44420311DD0E022C0E0E9D2
:100150003138E83138D0E022C0E0EB3138EA3138CC
:10016000E93138E83138D0E022C8CCC8C9CDC9CA95
:10017000CECACBCFCB22C0E0E5D05418B418027160
:100180004DE5D02408F5D0541824F8F8D0E022C06A
:10019000E0E5D054187002714DE5D024F8F5D0D0C8
:1001A000E022C0F0C0E0F5F0ECC0E0EDC0E0EEC051
:1001B000E0EFC0E03169E581F82410F58179101194
:1001C000B908F6D5F003198002D9F43169315831F4
:1001D00069743A311DE824F029F87A1008EA5403CA
:1001E00070047420311DEAD39950087420311D31F8
:1001F0001D8003E631387420311DDAE0E824F0F880
:100200007420311D747C311D7410C399F9FA08E60D
:10021000B420004005B47F004002742E311DDAEE98
:10022000747C311D740D311D740A311DE824F0F801
:10023000EC29FCED3400FDEE3400FEEF3400FFE568
:10024000F0600221BDE58124F0F5813169D0E0FF45
:10025000D0E0FED0E0FDD0E0FCD0E0D0F022C08EB7
:10026000C082C083E58E55E14510F58E85348285C8
:100270003583E0A3858234858335D083D082D08EC8
:1002800022C0A8C2AFC0E0C082C0833176E6FE08BB
:10029000E6FFEE4F7002714DEE240354FCFEEF3486
:1002A00000FFAA38AB398A828B83E0FCA3E0FDEC27
:1002B0004D7002714D8C828D83A3A3E0F9A3C3E03E
:1002C0009F701CE99E70188A828B8305A28C828D98
:1002D00083E0A305A2F0A305A2E005A2F0804750A9
:1002E0000FECFAEDFB8C828D82E0FCA3E0FD80BF79
:1002F0008A828B83EC2EFAF0A3ED3FFBF08A828B8F
:100300008305A28C828D83E0A305A2F0A305A2E061
:10031000A305A2F0A305A2E0A3C39E05A2F0A30536
:10032000A2E09F05A2F0853882853983A3A3E0C3AC
:100330009EF0A3E09FF0EDF618ECF6318FD083D05D
:1003400082D0E0D0A822C0A8C2AFD0A822C2AF906D
:1003500000A98582328583339003673122D0E03152
:1003600038D0E0313880FE0D0A2021212120706133
:060370006E6963204000ED
:00000001FF
