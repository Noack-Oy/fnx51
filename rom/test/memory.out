BOOTLOADER
> U
< U
> :020000050B00EE
< :020000050B00EE7B.\r\n
WRITE: memory.hex
> :100000007581509000B98582308583319000B1852B
< :100000007581509000B98582308583319000B1852B.\r\n
> :10001000823285833311A7538EE1438E1090000006
< :10001000823285833311A7538EE1438E1090000006.\r\n
> :100020008582388583397404F540F0A3E4F541F006
< :100020008582388583397404F540F0A3E4F541F006.\r\n
> :10003000A37542FF85064374FCF0A37406F0A3E4A5
< :10003000A37542FF85064374FCF0A37406F0A3E4A5.\r\n
> :10004000F0A3F0A374FCF0A37406F0A311C1B444B0
< :10004000F0A3F0A374FCF0A37406F0A311C1B444B0.\r\n
> :100050002611C68834893511C6E8A834A9357A0036
< :100050002611C68834893511C6E8A834A9357A0036.\r\n
> :100060007B00C030C031900266858230858331319B
< :100060007B00C030C031900266858230858331319B.\r\n
> :10007000AAD031D03080D5B4410811C6518931554C
< :10007000AAD031D03080D5B4410811C6518931554C.\r\n
> :10008000801BB4521411C6E8FCE9FD11C6E8FAE978
< :10008000801BB4521411C6E8FCE9FD11C6E8FAE978.\r\n
> :10009000FBECF8EDF9714E8004743F3125740D319D
< :10009000FBECF8EDF9714E8004743F3125740D319D.\r\n
> :1000A00025740A312580A5759852759ABF759B1ED7
< :1000A00025740A312580A5759852759ABF759B1ED7.\r\n
> :1000B000223099FDC299F599223098FDC298E599B0
< :1000B000223099FDC299F599223098FDC298E599B0.\r\n
> :1000C00022C030C03122C0E0ECC0E0EDC0E0E4F876
< :1000C00022C030C03122C0E0ECC0E0EDC0E0E4F876.\r\n
> :1000D000F9FAFB11C1C394304042B40A00400D94B8
< :1000D000F9FAFB11C1C394304042B40A00400D94B8.\r\n
> :1000E000114039240A54DFB410005030FCE8C4FD3C
< :1000E000114039240A54DFB410005030FCE8C4FD3C.\r\n
> :1000F00054F04CF8ED540FFCE9C4FD54F04CF9ED0C
< :1000F00054F04CF8ED540FFCE9C4FD54F04CF9ED0C.\r\n
> :10010000540FFCEAC4FD54F04CFAED540FFCEBC460
< :10010000540FFCEAC4FD54F04CFAED540FFCEBC460.\r\n
> :10011000FD54F04CFBED540F700280B7D0E0FDD0E1
< :10011000FD54F04CFBED540F700280B7D0E0FDD0E1.\r\n
> :10012000E0FCD0E022C032C03322C0E0E8C0E0E40E
< :10012000E0FCD0E022C032C03322C0E0E8C0E0E40E.\r\n
> :10013000F8936006312508E870F7D0E0F8D0E022A7
< :10013000F8936006312508E870F7D0E0F8D0E022A7.\r\n
> :10014000C43144C4C0E0540F2490D43440D444207B
< :10014000C43144C4C0E0540F2490D43440D444207B.\r\n
> :100150003125D0E022C0E0E93140E83140D0E02252
< :100150003125D0E022C0E0E93140E83140D0E02252.\r\n
> :10016000C0E0EB3140EA3140E93140E83140D0E0D5
< :10016000C0E0EB3140EA3140E93140E83140D0E0D5.\r\n
> :1001700022C8CCC8C9CDC9CACECACBCFCB22C0E019
< :1001700022C8CCC8C9CDC9CACECACBCFCB22C0E019.\r\n
> :10018000E5D05418B418029174E5D02408F5D05481
< :10018000E5D05418B418029174E5D02408F5D05481.\r\n
> :100190001824F8F8D0E022C0E0E5D054187002919D
< :100190001824F8F8D0E022C0E0E5D054187002919D.\r\n
> :1001A00074E5D024F8F5D0D0E022C0F0C0E0F5F03E
< :1001A00074E5D024F8F5D0D0E022C0F0C0E0F5F03E.\r\n
> :1001B000ECC0E0EDC0E0EEC0E0EFC0E03171E58101
< :1001B000ECC0E0EDC0E0EEC0E0EFC0E03171E58101.\r\n
> :1001C000F82410F581791011C108F6D5F0031980D3
< :1001C000F82410F581791011C108F6D5F0031980D3.\r\n
> :1001D00002D9F4317131603171743A3125E824F07B
< :1001D00002D9F4317131603171743A3125E824F07B.\r\n
> :1001E00029F87A1008EA5403700474203125EAD300
< :1001E00029F87A1008EA5403700474203125EAD300.\r\n
> :1001F0009950087420312531258003E63140742060
< :1001F0009950087420312531258003E63140742060.\r\n
> :100200003125DAE0E824F0F874203125747C3125BA
< :100200003125DAE0E824F0F874203125747C3125BA.\r\n
> :100210007410C399F9FA08E6B420004005B47F00D1
< :100210007410C399F9FA08E6B420004005B47F00D1.\r\n
> :100220004002742E3125DAEE747C3125740D3125AF
< :100220004002742E3125DAEE747C3125740D3125AF.\r\n
> :10023000740A3125E824F0F8EC29FCED3400FDEED9
< :10023000740A3125E824F0F8EC29FCED3400FDEED9.\r\n
> :100240003400FEEF3400FFE5F0600221C5E58124B3
< :100240003400FEEF3400FFE5F0600221C5E58124B3.\r\n
> :10025000F0F5813171D0E0FFD0E0FED0E0FDD0E0DC
< :10025000F0F5813171D0E0FFD0E0FED0E0FDD0E0DC.\r\n
> :10026000FCD0E0D0F022C08EC082C083E58E55E184
< :10026000FCD0E0D0F022C08EC082C083E58E55E184.\r\n
> :100270004510F58E853482853583E0A385823485EB
< :100270004510F58E853482853583E0A385823485EB.\r\n
> :100280008335D083D082D08E22C0A8C2AFC0E0C058
< :100280008335D083D082D08E22C0A8C2AFC0E0C058.\r\n
> :1002900082C083317EE6FE08E6FFEE4F7002917465
< :1002900082C083317EE6FE08E6FFEE4F7002917465.\r\n
> :1002A000EE240354FCFEEF3400FFAA38AB398A82F7
< :1002A000EE240354FCFEEF3400FFAA38AB398A82F7.\r\n
> :1002B0008B83E0FCA3E0FDEC4D700291748C828D89
< :1002B0008B83E0FCA3E0FDEC4D700291748C828D89.\r\n
> :1002C00083A3A3E0F9A3C3E09F701CE99E70188A82
< :1002C00083A3A3E0F9A3C3E09F701CE99E70188A82.\r\n
> :1002D000828B8305A28C828D83E0A305A2F0A30507
< :1002D000828B8305A28C828D83E0A305A2F0A30507.\r\n
> :1002E000A2E005A2F08047500FECFAEDFB8C828D66
< :1002E000A2E005A2F08047500FECFAEDFB8C828D66.\r\n
> :1002F00082E0FCA3E0FD80BF8A828B83EC2EFAF0C3
< :1002F00082E0FCA3E0FD80BF8A828B83EC2EFAF0C3.\r\n
> :10030000A3ED3FFBF08A828B8305A28C828D83E074
< :10030000A3ED3FFBF08A828B8305A28C828D83E074.\r\n
> :10031000A305A2F0A305A2E0A305A2F0A305A2E015
< :10031000A305A2F0A305A2E0A305A2F0A305A2E015.\r\n
> :10032000A3C39E05A2F0A305A2E09F05A2F0853815
< :10032000A3C39E05A2F0A305A2E09F05A2F0853815.\r\n
> :1003300082853983A3A3E0C39EF0A3E09FF0EDF68E
< :1003300082853983A3A3E0C39EF0A3E09FF0EDF68E.\r\n
> :1003400018ECF63197D083D082D0E0D0A822C0A894
< :1003400018ECF63197D083D082D0E0D0A822C0A894.\r\n
> :10035000C2AFC0E0C082C083317EE82403F8E6FF6C
< :10035000C2AFC0E0C082C083317EE82403F8E6FF6C.\r\n
> :1003600018E6FE18E6F918E6F8EE4F70029174E907
< :1003600018E6FE18E6F918E6F8EE4F70029174E907.\r\n
> :10037000B54104E8B5400050029174E9B54304E882
< :10037000B54104E8B5400050029174E9B54304E882.\r\n
> :10038000B5420040029174EE240354FCFEEF3400A9
< :10038000B5420040029174EE240354FCFEEF3400A9.\r\n
> :10039000FFAA38AB398A828B83E0FCA3E0FDEC4DE9
< :10039000FFAA38AB398A828B83E0FCA3E0FDEC4DE9.\r\n
> :1003A000601CC3ED997006EC9870029174500FECCC
< :1003A000601CC3ED997006EC9870029174500FECCC.\r\n
> :1003B000FAEDFB8C828D83E0FCA3E0FD80E0EBB5E1
< :1003B000FAEDFB8C828D83E0FCA3E0FD80E0EBB5E1.\r\n
> :1003C0003906EAB5380280308A828B83A3A3E02AFB
< :1003C0003906EAB5380280308A828B83A3A3E02AFB.\r\n
> :1003D000C0E0A3E03BF583D082E9B58317E8B5829E
< :1003D000C0E0A3E03BF583D082E9B58317E8B5829E.\r\n
> :1003E000138A828B83A3A3E02EF0A3E03FF0EAF808
< :1003E000138A828B83A3A3E02EF0A3E03FF0EAF808.\r\n
> :1003F000EBF9801C5002917488828983ECF0A3EDA4
< :1003F000EBF9801C5002917488828983ECF0A3EDA4.\r\n
> :10040000F0A3EEF0A3EFF08A828B83E8F0A3E9F08B
< :10040000F0A3EEF0A3EFF08A828B83E8F0A3E9F08B.\r\n
> :1004100088828983A3A3E028FAA3E039F5838A823E
< :1004100088828983A3A3E028FAA3E039F5838A823E.\r\n
> :10042000EDB58332ECB5822E8882898305A28C8259
< :10042000EDB58332ECB5822E8882898305A28C8259.\r\n
> :100430008D83E0A305A2F0A305A2E0A305A2F0A38B
< :100430008D83E0A305A2F0A305A2E0A305A2F0A38B.\r\n
> :1004400005A2E0A3FA05A2E02AF0A305A2E0FB05BD
< :1004400005A2E0A3FA05A2E02AF0A305A2E0FB05BD.\r\n
> :10045000A2E03BF080045002917485388285398394
< :10045000A2E03BF080045002917485388285398394.\r\n
> :10046000A3A3E02EF0A3E03FF03197D083D082D059
< :10046000A3A3E02EF0A3E03FF03197D083D082D059.\r\n
> :10047000E0D0A822C2AF9000B1858232858333904C
< :10047000E0D0A822C2AF9000B1858232858333904C.\r\n
> :10048000048E312AD0E03140D0E0314080FE0D0AA8
< :10048000048E312AD0E03140D0E0314080FE0D0AA8.\r\n
> :0D049000202121212070616E696320400051
< :0D049000202121212070616E696320400051.\r\n
> :00000001FF
< :00000001FF.\r\n
RESET
> D0;10;
00000000: 04 00 fc 06  00 00 fc 06  34 50 04 60  5a 17 14 8e  |........4P.`Z...|
> A0123;
0004
> D0000;10;
00000000: 28 01 d8 05  00 00 fc 06  34 50 04 60  5a 17 14 8e  |(.......4P.`Z...|
> D0128;10;
00000128: 00 00 d8 05  82 00 40 31  84 50 70 11  d1 02 f6 44  |......@1.Pp....D|
> A0312;
0128
> D0000;10;
00000000: 3c 04 c4 02  00 00 fc 06  34 50 04 60  5a 17 14 8e  |<.......4P.`Z...|
> D043c;10;
0000043c: 00 00 c4 02  84 08 77 02  83 ab 78 61  48 00 5e 22  |......w...xaH.^"|
> A01c4;
043c
> D0000;10;
00000000: 00 06 00 01  00 00 fc 06  34 50 04 60  5a 17 14 8e  |........4P.`Z...|
> R0128;0312;

> D0000;10;
00000000: 28 01 14 04  00 00 fc 06  34 50 04 60  5a 17 14 8e  |(.......4P.`Z...|
> D0128;10;
00000128: 00 06 14 03  82 00 40 31  84 50 70 11  d1 02 f6 44  |......@1.Pp....D|
> R043c;01c4;

> D0128;10;
00000128: 00 00 d8 05  82 00 40 31  84 50 70 11  d1 02 f6 44  |......@1.Pp....D|
> R0004;0100;

> D0000;10;
00000000: 04 00 d8 06  28 01 00 01  34 50 04 60  5a 17 14 8e  |....(...4P.`Z...|
> R0104;0023;

> D0000;10;
00000000: 04 00 fc 06  00 00 fc 06  34 50 04 60  5a 17 14 8e  |........4P.`Z...|
